### DISCUSSION TITLE
Default sonatype uri?

### DISCUSSION THREAD

**Discussion by Quafadas** [com-lihaoyi/mill#2927] [General]
It took me a lot longer, than it should have done, to realise, that the error message said 403;
https://oss.sonatype.org/service/local/staging/profiles

instead of the new
https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/

I would assume (but checking), there is a good reason the legacy sonatype url remains the default? Otherwise, I would happily put up the teeniest of tiniest PR's, changing it...

**Comment by lefou** [comment]
To quote myself on that topic (source: https://github.com/com-lihaoyi/mill/pull/1687#issuecomment-1019548146)

> It's a pity Sonatype hasn't some dispatching facade for it, but I'm not going to change these defaults. As these are just defaults, they can and should be overridden by the user. It's probably questionable that we have a default at all, but changing it does not make much sense, as It will break lots of workflows including Mills one, and it will still be the wrong URL for other users.

and (source: https://github.com/com-lihaoyi/mill/pull/1687#issuecomment-1019548466)

> We may rethink, once the legacy host is shut down and _all_ user are migrated to the new one.

**Reply by Quafadas** [reply]
A very reasonable response - thanks. I shall stand that PR down :-). 

I may propose a little update to this note;

>  Since Feb. 2021 any new Sonatype accounts have been created on s01.oss.sonatype.org, so you’ll want to ensure you set the relevant URIs to match.

Including a further hint, that the classic symptom of hitting this mismatched url problem, is a 403 response to the publish task?

**Reply by lefou** [reply]
Improving the docs sounds like a good idea. Looks like you got a second change to open a "tiny" PR. ;)
